raiden_libs/network/matrix/client.py

```diff
@@ -259,11 +236,11 @@ def _sync(self, timeout_ms=30000):
 
             for event in sync_room["state"]["events"]:
                 event['room_id'] = room_id
-                room._process_state_event(event)
+                self.call(room._process_state_event, event)
 
             for event in sync_room["timeline"]["events"]:
                 event['room_id'] = room_id
-                room._put_event(event)
+                self.call(room._put_event, event)
```

Comment:
So the only downside I can see here is that `_put_event` calls all room listeners in a loop. But for our use case I guess it's fine since we usually only have one listener per room anyway.